### PR TITLE
Create codeql-analysis.yaml

### PR DESCRIPTION
**Pull request by sdunnack**
Initial commit, will iterate on errors

### CODE REVIEW

File: .github/workflows/codeql-analysis.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    #     echo "CODEQL_PYTHON=$(which python)" >> $GITHUB_ENV
+    
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        setup-python-dependencies: false
+        queries: +security-and-quality,octo-org/python-qlpack/show_ifs.ql@main
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
```
Comment (margaretpearce) [inline .github/workflows/codeql-analysis.yaml:68]: I think we should try adding:
```
        external-repository-token: ${{ secrets.ACCESS_TOKEN }}
```

Comment (margaretpearce) [reply .github/workflows/codeql-analysis.yaml:68]: Explanation: we are trying to pull in an external repo--

```
 Checking out octo-org/python-qlpack
  /usr/bin/git clone https://github.com/octo-org/python-qlpack /home/runner/work/_temp/octo-org/python-qlpack/main
  Cloning into '/home/runner/work/_temp/octo-org/python-qlpack/main'...
  fatal: could not read Username for 'https://github.com': No such device or address
  Error: The process '/usr/bin/git' failed with exit code 128
```

**Comment by mjs2600** [comment]
@sdunnack can this be closed? There's a codeql file there now. https://github.com/thorn-oss/perception/blob/main/.github/workflows/codeql-analysis.yml

**Comment by mjs2600** [comment]
Closing, but we can reopen it if there's more to do here.

**Pull request closed without merging**